core/src/mindustry/world/meta/BlockStats.java

```diff
@@ -38,8 +38,16 @@ public void add(BlockStat stat, Liquid liquid, float amount, boolean perSecond){
     }
 
     public void add(BlockStat stat, Attribute attr){
+        add(stat, attr, +1);
+    }
+
+    public void add(BlockStat stat, Attribute attr, int sign){
         for(Block block : Vars.content.blocks()){
-            if(!block.isFloor() || block.asFloor().attributes.get(attr) <= 0) continue;
```

Comment:
The attribute sign check can be written (without the switch) as `if(!block.isFloor() || !Mathf.equal(Math.signum(block.asFloor().attributes.get(attr)), sign)) continue;`